src/java.base/aix/classes/sun/nio/fs/AixFileSystemProvider.java

```diff
@@ -68,8 +69,9 @@ public <V extends FileAttributeView> V getFileAttributeView(Path obj,
                                                                 LinkOption... options)
     {
         if (type == UserDefinedFileAttributeView.class) {
-            return !supportsUserDefinedFileAttributeView(obj) ? null :
-                (V) new AixUserDefinedFileAttributeView(UnixPath.toUnixPath(obj), Util.followLinks(options));
+            return supportsUserDefinedFileAttributeView(obj) ?
+                (V) new AixUserDefinedFileAttributeView(UnixPath.toUnixPath(obj), Util.followLinks(options))
+                : null;
```

Comment:
Can you try this:
```
if (type == UserDefinedFileAttributeView.class) { 
    UnixPath file = UnixPath.toUnixPath(obj);
    return supportsUserDefinedFileAttributeView(file) ? ..
```

That will remove the cast/check from supportsUserDefinedFileAttributeView and also ensure that ProviderMismatchException is thrown if called with a Path from a different provider.